### DISCUSSION TITLE
How do I access a dynamic CDN

### DISCUSSION THREAD

**Discussion by ljj0915** [vercel/next.js#66595] [Ideas]
### Goals

1.It is hoped that the CDN address of static resources can be dynamically changed

### Non-Goals

1.
2.
3.

### Background

1. By default, when you visit a nextjs page, static resources such as related Js and Css are accessed through paths such as [current server address]/.next/static.
2.  In other words, resources follow the server. But I want this part of the resource deployed on the CDN.
3. Although there is an assetPrefix configuration item that can replace the domain name of the resource, what if my CDN server can provide the CDN service dynamically? That is, the domain name is not fixed, and the optimal CDN address will be returned in real time according to the user's request.
4.  In this case, how can nextjs access dynamic CDN

### Proposal

I hope I can expose an api that allows me to replace the request path of the resource myself, so that I can change the obtaining path of the static resource every time I request a page

**Comment by wwwdepot** [comment]
GeoDNS. What you're trying to do should be done on DNS, not on the app server.
